clang/include/clang/Basic/LangOptions.h

```diff
@@ -190,10 +190,22 @@ class LangOptions : public LangOptionsBase {
                   /// (More C compatible than offset)
   };
 
+  enum CheriCapIntConversion {
+    CapInt_Explicit, /// Require explicit casts for pointer -> capability and
+                     /// integer -> capability pointer
+    CapInt_Strict,  /// All conversions (including (u)intcap<->int need explicit
+                    /// casts)
```

Comment:
It's unclear at first glance what the difference is. My belief is that Strict is even stricter than Explicit given it also stops (u)intcap_t casts? If so I think it would be better to put it first, then the enum is in order of strictness from most to least (well, Address and Relative are the same). Or put them in the reverse order so you can refer back to weaker ones, as you want to do in cheri_cap_to_int_EQ. Though I'm unconvinced stopping (u)intcap<->int casts is a good idea, especially for purecap....

Reply:
My thinking was that the strict mode is something you can turn on if you are paranoid about accidental tag loss.
I have seen one or two cases where this would have helped, but it's generally going to have an extremely low signal-to-noise ratio.

Could potentially be useful for something like this:
```
uintptr_t where_hass_my_tag_gone(uintptr_t maybe_tagged) {
    long l = maybe_tagged + 1; // long is the same as uintptr_t right???
    // Some more code
    return l;
}
```

It could possibly be useful to enforce compiler errors so that uintptr_t is only used for "maybe-pointer/pointer-sized" values rather than "default register size".
But this is a low-priority future task, I just thought I'd reserve the enum constants now.

Changing the order sounds like a good suggestion, will do.

Reply:
Yeah, that's something to catch, but the other way round isn't IMO. And I really don't want to be forcing the __cheri_foo casts on either direction, as currently we (ab)use uintptr_t to avoid those in a bunch of places.